sklearn/ensemble/_hist_gradient_boosting/tests/test_loss.py

```diff
@@ -70,20 +72,28 @@ def test_derivatives(loss, x0, y_true):
     x0 = np.array([x0], dtype=Y_DTYPE).reshape(1, 1)
     get_gradients, get_hessians = get_derivatives_helper(loss)
 
-    def func(x):
-        return loss.pointwise_loss(y_true, x)
+    def func(x: np.ndarray) -> np.ndarray:
+        if isinstance(loss, _LOSSES['binary_crossentropy']):
+            # Subtract a constant term such that the binary cross entropy
+            # has its minimum at zero. This only works if 0 < y_true < 1.
+            actual_min = loss.pointwise_loss(y_true, logit(y_true))
+            return (loss.pointwise_loss(y_true, x) - actual_min)
+        else:
+            return loss.pointwise_loss(y_true, x)
 
-    def fprime(x):
+    def fprime(x: np.ndarray) -> np.ndarray:
```

Comment:
I don't feel like we need the annotations anymore. Especially given that these expect arrays with a single element. Newton  would break otherwise (I think). Maybe a comment is enough?